tnnt/views.py

```diff
@@ -11,3 +26,310 @@ class AboutView(TemplateView):
+    # don't require that.
+    # This returns the Player associated with oth_id.
+    def clan_admin_other_member_checks(self, request, player, ctx, oth_id):
+        # other player must exist
+        print(oth_id)
+        try:
+            otherplayer = Player.objects.get(id=oth_id)
+        except Player.DoesNotExist:
+            impossible('%s attempted to do clan admin stuff with nonexistent id %d'
+                       % (player.name, oth_id))
+            ctx['errmsg'] = 'The player was not found.'
+            return None
+
+        # player must be in a clan
+        if player.clan is None:
+            impossible('%s attempted to do clan admin stuff without being in a clan'
+                       % (player.name))
+            ctx['errmsg'] = "You can't do that if you're not in a clan!"
+            return None
+
+        # player must be an admin of their clan
+        if not player.clan_admin:
+            ctx['errmsg'] = 'Only admins can do this'
+            return None
+
+        return otherplayer
+
+    # Helper function triggered when "Rescind" is clicked on an invited player
+    def rescind_invite(self, request, player, ctx):
+        # TODO? all of these helpers should test that the key is in POST, and if
+        # not, log an error and return
+        rescindee_id = request.POST['rescind_id']
+        rescindee = self.clan_admin_other_member_checks(request, player, ctx, rescindee_id)
+        if rescindee is None:
+            return
+
+        # if we passed checks, we're good to rescind the invite
+        rescindee.invites.remove(player.clan)
+
+    # Helper function triggered when "Make Admin" is clicked on a clan member
+    def make_admin(self, request, player, ctx):
+        new_admin_id = request.POST['kick_or_admin_id']
+        new_admin = self.clan_admin_other_member_checks(request, player, ctx, new_admin_id)
+        if new_admin is None:
+            return
+
+        # other player must be in this clan
+        if player.clan != new_admin.clan:
+            impossible('%s attempted to do make %s (not in their clan) an admin'
+                       % (player.name, otherplayer.name))
+            ctx['errmsg'] = otherplayer.name + ' is not in your clan'
+            return
+
+        # can't make someone an admin who's already an admin
+        if new_admin.clan_admin:
+            ctx['errmsg'] = '%s is already a clan admin!' % (new_admin.name)
+            return
+
+        # if we passed checks, we're good to make this person an admin
+        new_admin.clan_admin = True
+        new_admin.save()
+
+    # Helper function triggered when "Kick" is clicked on a clan member
+    def kick_member(self, request, player, ctx):
+        kickee_id = request.POST['kick_or_admin_id']
+        kickee = self.clan_admin_other_member_checks(request, player, ctx, kickee_id)
+        if kickee is None:
+            return
+
+        # other player must be in this clan
+        if player.clan != kickee.clan:
```

Comment:
I would use identity rather than equality here.

Reply:
Using `if player.clan is not kickee.clan:` results in the admin not being able to kick anyone.